src/MediaDisplay.jsx

```diff
@@ -21,50 +21,28 @@ export default class MediaDisplay extends React.Component {
         this.className = 'jux-media-video';
         this.state = {duration: null};
     }
-    renderVideo(data) {
-        let url = null;
-        if (data.host === 'youtube') {
-            url = 'https://www.youtube.com/watch?v=' + data.source;
-        } else if (data.host === 'vimeo') {
-            url = 'https://www.vimeo.com/watch?v=' + data.source;
-        } else {
-            url = data.source;
-        }
-
-        return <ReactPlayer
-            id={this.id}
-            ref={(ref) => this.player = ref}
-            width={480}
-            height={360}
-            playing={this.props.playing}
-            url={url}
-            onDuration={this.onDuration.bind(this)}
-            onStart={this.onVidStart.bind(this)}
-            progressFrequency={100}
-        />;
+    componentDidUpdate(props, state) {
+        this.generatePlayers(props.data);
     }
-    onVidStart() {
-        const vid = getCurrentItem(this.props.data, this.props.time);
-        if (vid.annotationStartTime && this.state.duration) {
-            const percentage = vid.annotationStartTime / this.state.duration;
-            this.player.seekTo(percentage);
+    generatePlayers(mediaTrack) {
+        this.players = [];
```

Comment:
Yeah this is called on each `onSpineProgress` call every 20ms since it updates the time prop.. I didn't know where else to put this and in practice React seems to be keeping the SecondaryPlayer instances in place when they don't change - otherwise this wouldn't work at all. Could definitely be re-thought to work in a different way. I couldn't put this in the constructor because the media element data isn't there until we load everything from the Mediathread API.